cli/azd/pkg/commands/pipeline/pipeline_manager.go

```diff
@@ -200,26 +203,31 @@ func (manager *PipelineManager) Configure(ctx context.Context) error {
 		// changed from "az-cli" to "az-dev"
 		manager.PipelineServicePrincipalName = fmt.Sprintf("az-dev-%s", time.Now().UTC().Format("01-02-2006-15-04-05"))
 	}
+
 	inputConsole.Message(ctx, fmt.Sprintf("Creating or updating service principal %s.\n", manager.PipelineServicePrincipalName))
-	credentials, err := azCli.CreateOrUpdateServicePrincipal(
-		ctx,
-		manager.Environment.GetSubscriptionId(),
-		manager.PipelineServicePrincipalName,
-		manager.PipelineRoleName)
-	if err != nil {
-		return fmt.Errorf("failed to create or update service principal: %w", err)
+	var credentials json.RawMessage = nil
```

Comment:
Maybe refactor out `GetOrCreateServicePrincipal` given how long `Configure` is..

Reply:
nit?